src/blas/backends/cublas/cublas_extensions.cpp

```diff
@@ -73,44 +68,50 @@ void gemm_ext(cl::sycl::queue &queue, transpose transa, transpose transb, offset
     throw backend_unsupported_exception();
 }
 
-void gemm_ext(cl::sycl::queue &queue, transpose transa, transpose transb, std::int64_t m,
-              std::int64_t n, std::int64_t k, float alpha, cl::sycl::buffer<float, 1> &a,
-              std::int64_t lda, cl::sycl::buffer<float, 1> &b, std::int64_t ldb, float beta,
-              cl::sycl::buffer<float, 1> &c, std::int64_t ldc) {
-    throw backend_unsupported_exception();
-}
-
-void gemm_ext(cl::sycl::queue &queue, transpose transa, transpose transb, std::int64_t m,
-              std::int64_t n, std::int64_t k, double alpha, cl::sycl::buffer<double, 1> &a,
-              std::int64_t lda, cl::sycl::buffer<double, 1> &b, std::int64_t ldb, double beta,
-              cl::sycl::buffer<double, 1> &c, std::int64_t ldc) {
-    throw backend_unsupported_exception();
+template <typename Func, typename T_AB, typename T_C, typename DATATYPE_AB, typename DATATYPE_C>
+inline void gemm_ext(Func func, DATATYPE_AB DT_AB, DATATYPE_C DT_C, cl::sycl::queue &queue,
```

Comment:
To anticipate future functionality, could you use different template type for A and B matrices?

Reply:
Makes sense. Will update.

Reply:
Template parameters are updated.